eth/beacon/state_machines/forks/serenity/epoch_transitions.py

```diff
@@ -0,0 +1,224 @@
+        WHERE
+            a.slot between `now` and `1 epoch ago` AND
+            a.justified_slot = justified_slot AND
+            a.epoch_boundary_root = `1 epoch ago`
+        GROUP BY v.index
+    """
+
+    EPOCH_LENGTH = config.EPOCH_LENGTH
+    MAX_DEPOSIT = config.MAX_DEPOSIT
+    LATEST_BLOCK_ROOTS_LENGTH = config.LATEST_BLOCK_ROOTS_LENGTH
+
+    current_epoch_attestations = tuple(
+        attestation
+        for attestation in state.latest_attestations
+        if state.slot - EPOCH_LENGTH <= attestation.data.slot < state.slot
+    )
+    previous_epoch_attestations = tuple(
+        attestation
+        for attestation in state.latest_attestations
+        if state.slot - 2 * EPOCH_LENGTH <= attestation.data.slot < state.slot - EPOCH_LENGTH
+    )
+
+    previous_epoch_justified_attestations = tuple(
+        attestation
+        for attestation in current_epoch_attestations + previous_epoch_attestations
+        if attestation.justified_slot == state.previous_justified_slot
+    )
+
+    previous_epoch_boundary_root = get_block_root(
+        state,
+        state.slot - 2 * EPOCH_LENGTH,
+        LATEST_BLOCK_ROOTS_LENGTH)
+    previous_epoch_boundary_attestations = tuple(
+        attestation
+        for attestation in previous_epoch_justified_attestations
+        if attestation.epoch_boundary_root == previous_epoch_boundary_root
+    )
+
+    sets_of_previous_epoch_boundary_participants = tuple(
+        frozenset(get_attestation_participants(
+            state,
+            attestation.data.slot,
+            attestation.data.shard,
+            attestation.participation_bitfield,
+            EPOCH_LENGTH,
+        ))
+        for attestation in previous_epoch_boundary_attestations
+    )
+    previous_epoch_boundary_attester_indices = (
+        tuple()
+        if len(sets_of_previous_epoch_boundary_participants) == 0
+        else frozenset.union(*sets_of_previous_epoch_boundary_participants)
+    )
+    previous_epoch_boundary_attesting_balance = Gwei(sum(
+        get_effective_balance(state, index, MAX_DEPOSIT)
+        for index in previous_epoch_boundary_attester_indices
+    ))
+
+    current_epoch_boundary_root = get_block_root(
+        state,
+        state.slot - EPOCH_LENGTH,
+        LATEST_BLOCK_ROOTS_LENGTH)
+
+    current_epoch_boundary_attestations = tuple(
+        attestation
+        for attestation in current_epoch_attestations
+        if attestation.epoch_boundary_root == current_epoch_boundary_root and
+        attestation.data.justified_slot == state.justified_slot
+    )
+
+    sets_of_current_epoch_boundary_participants = tuple(
+        frozenset(get_attestation_participants(
+            state,
+            attestation.data.slot,
+            attestation.data.shard,
+            attestation.participation_bitfield,
+            EPOCH_LENGTH,
+        ))
+        for attestation in current_epoch_boundary_attestations
+    )
+
+    current_epoch_boundary_attester_indices = (
+        tuple()
```

Comment:
Same here with the type mismatch.  Maybe this should be an empty `frozenset()` to keep things homogeneous.

Reply:
Will fix